R/po_compile.R

```diff
@@ -0,0 +1,62 @@
+#' Compile `.po` files to `.mo`
+#'
+#' This function compiles the plain text `.po` files that translators work with
+#' into the binary `.mo` files that are installed with packages and used for
+#' live translations.
+#'
+#' @param dir Path to package root directory.
+#' @param package Name of package. If not supplied, read from `DESCRIPTION`.
+#' @param lazy If `TRUE`, only `.mo` functions that are older than `.po`
+#'   files be updated
+#' @param verbose If `TRUE`, print information as it goes.
+po_compile = function(dir = ".", package = NULL, lazy = TRUE, verbose = TRUE) {
+  po_metadata <- get_po_metadata(dir = dir, package = package)
+  dir_create(dirname(po_metadata$mo))
+
+  if (lazy) {
+    po_metadata <- po_metadata[is_outdated(po_metadata$po, po_metadata$mo)]
+  } else {
+    # Clear out all older translations
+    mo_dirs <- dir(file.path(dir, "inst", "po"), full.names = TRUE)
+    to_delete <- mo_dirs[!basename(mo_dirs) %in% c(po_metadata$language, "en@quot")]
+
+    for (dir in to_delete) {
+      if (verbose) messagef(
+        "Found a compiled translation for %s at %s, but no corresponding .po; deleting",
```

Comment:
This will be a bit noisy if several .mo have failed, especially since `dirname(to_delete)` should all be the same right?

Should we move the `verbose` message outside the loop & announce the deletions all up front?

```
if (verbose) messagef(
  "Found compiled translations  in %s without corresponding .po: %s. Deleting.",
  dirname(to_delete[1L]), toString(basename(to_delete))
)
for (dir in to_delete) unlink(dir, recursive = TRUE)
```

Reply:
I think this should be pretty unusual — you'll only see this if you had a `.po`, compiled it, and then later deleted it.